numpy/core/_methods.py

```diff
@@ -56,14 +56,25 @@ def _any(a, axis=None, dtype=None, out=None, keepdims=False):
 def _all(a, axis=None, dtype=None, out=None, keepdims=False):
     return umr_all(a, axis, dtype, out, keepdims)
 
-def _count_reduce_items(arr, axis):
-    if axis is None:
-        axis = tuple(range(arr.ndim))
-    if not isinstance(axis, tuple):
-        axis = (axis,)
-    items = 1
-    for ax in axis:
-        items *= arr.shape[mu.normalize_axis_index(ax, arr.ndim)]
+def _count_reduce_items(arr, axis, keepdims=False, where=True):
+    if isinstance(where, bool):
+        if not where:
+            # no elements in slice
+            return 0
+        # no boolean mask given, calculate items according to axis
+        if axis is None:
+            axis = tuple(range(arr.ndim))
+        elif not isinstance(axis, tuple):
+            axis = (axis,)
+        items = 1
+        for ax in axis:
+            items *= arr.shape[mu.normalize_axis_index(ax, arr.ndim)]
+    else:
+        # guarded to protect circular imports
```

Comment:
Worth a TODO here about this sum bring more expensive than needed for when where is broadcast along a non-reduction axis.

Reply:
Good point, added a TODO note.